activerecord/lib/active_record/persistence.rb

```diff
@@ -143,7 +143,9 @@ def insert(attributes, returning: nil, unique_by: nil, record_timestamps: nil)
       #     { id: 2, title: "Eloquent Ruby" }
       #   ])
       def insert_all(attributes, returning: nil, unique_by: nil, record_timestamps: nil)
-        InsertAll.new(self, attributes, on_duplicate: :skip, returning: returning, unique_by: unique_by, record_timestamps: record_timestamps).execute
+        if attributes.any?
+          InsertAll.new(self, attributes, on_duplicate: :skip, returning: returning, unique_by: unique_by, record_timestamps: record_timestamps).execute
+        end
```

Comment:
```suggestion
        return if attributes.empty?
        InsertAll.new(self, attributes, on_duplicate: :skip, returning: returning, unique_by: unique_by, record_timestamps: record_timestamps).execute
```

It's a bit of a nitpick, but I'm always a bit uneasy with `Array#any?` being used as a shortcut for `!array.empty?`, it's not strictly equivalent because `[false, nil].any? # => false`.

In this case all the array elements should be arrays themselves, so valid input can't possibly be falsy, but someone may pass some invalid input have end up with a weird behavior.